### ISSUE TITLE
Mautic started redirecting to non-existing HTTPS after cache clear

### ISSUE THREAD

**Issue by Blackclaws** [mautic/mautic#11162]
### Mautic Version

4.2.x series

### PHP version

7.4.28

### What happened?

Reopened an old bug: https://github.com/mautic/mautic/issues/7943

The issue happens no matter the browser and also happens with console clients such as curl/wget etc. You get a 301 response to the https site.

### How can we reproduce this issue?

Step 1: Install Mautic
Step 2: Set https as the frontend URL scheme (this happens when you run it behind an SSL terminator/reverse proxy)
Step 3: Delete Mautic Cache rm -rf var/cache
Step 4: Mautic is now broken

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

**Comment by Blackclaws** [comment]
A workaround is by the way: 

- Change site url in config/local.php
- Rebuild cache
- Mautic works again, you can change the site url again.

**Comment by Blackclaws** [comment]
I think mautic is missing something like overwriteprotocol, a setting that nextcloud has in its config.php that overwrites the protocol mautic uses to generate its pages even when it itself is being accessed over http

**Comment by RCheesley** [comment]
Hi @Blackclaws glad you found a workaround - would you be open to researching if we could make this happen in Mautic and make a PR?

**Comment by jos0405** [comment]
Hi,
We experienced the same issue.
The solution is to add
`$_SERVER['HTTPS'] = 'on';`
after the opening php tag in index.php.
I'm not sure if Mautic is the culprit.
Can you plz check if it works for you?

**Comment by Blackclaws** [comment]
> Hi, We experienced the same issue. The solution is to add `$_SERVER['HTTPS'] = 'on';` after the opening php tag in index.php. I'm not sure if Mautic is the culprit. Can you plz check if it works for you?

I'm not sure if modifying mautics source code is really a "solution" in as much as its also a workaround. Mautic should not default to redirecting http to https traffic based on the site url, and especially not only after the cache is cleared.

Settings `$_SERVER['HTTPS'] = 'on';` as far as I can tell just forces php to tell mautic that indeed it was called via https. Which it wasn't in this case. 

>Hi @Blackclaws glad you found a workaround - would you be open to researching if we could make this happen in Mautic and make a PR?

I'll have to check whether my schedule allows me to deliver a fix for this as I'm not familiar with the mautic codebase at all and wouldn't really know where to start.

**Comment by jos0405** [comment]
Thank you. Let's leave this bug open.

**Comment by jos0405** [comment]
Just a comment, if you use http instead of https, many tracking features won't work properly.

**Comment by Blackclaws** [comment]
> Just a comment, if you use http instead of https, many tracking features won't work properly.

Does mautic itself disable some features? From outside mautic _is_ accessed via https. It just sits behind a reverse proxy that communicates over http with it (Kubernetes Cluster with Ingress).

**Comment by jos0405** [comment]
Hi, to me this looks like a routing issue to the containers, not a Mautic bug.

**Comment by Blackclaws** [comment]
> Hi, to me this looks like a routing issue to the containers, not a Mautic bug.

How exactly? Its quite normal to route to the actual containers via http and have ssl termination at the ingress. You need to correctly set the external address in mautic so that its links point to https sites instead of http. The fact that mautic itself performs an https redirect on incoming traffic _should not_ be dependent on the base url being https or not. The incoming traffic in such a setup will _always_ be coming via http from the view of the mautic container, while for external users they are using an https site.

**Comment by pdxrlk** [comment]
> 

THANK YOU.  Worked like a charm.

**Comment by alexander-nitsche-governikus** [comment]
Had the same problem today with Mautic v5.2.3 and it was due to a reverse proxy with SSL termination before the web server. Mautic then repeatedly redirects to https:// as it is configured to do, but it keeps getting a http:// from the proxy. We solved the problem by forwarding requests from the proxy to the web server on port 443 instead of 80. For example, in the local docker HAProxy v3.1.5 configuration we have replaced
```
backend webserver
    server webserver-01 webserver:80
```
with
```
backend webserver
    server webserver-01 webserver:443 ssl verify none
```
